Add bifrost aa1 to master

kleberbaum opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently there is no self generating GraphQL API like wagtail-grapple. #11 


**What is the new behavior (if this is a feature change)?**
- Now there is a self generating GraphQL API like wagtail-grapple. Closes #11 


**Other information**:
- Ref: https://pypi.org/project/graphql-core/
- Ref: https://pypi.org/project/graphene-django/
- Ref: https://pypi.org/project/graphene-django-optimizer/
- Ref: https://pypi.org/project/django-graphql-jwt/
- Ref: https://pypi.org/project/channels/
- Ref: https://pypi.org/project/asgi_redis/
- Ref: https://pypi.org/project/graphql-ws/

kleberbaum committed:
Add and configure bifrost requirements

Graphql API
graphql-core==2.2.1
Ref: https://pypi.org/project/graphql-core/
graphene-django==2.12.1
Ref: https://pypi.org/project/graphene-django/
graphene-django-optimizer==0.6.2
Ref: https://pypi.org/project/graphene-django-optimizer/

Authentication and Authorization
django-graphql-jwt==0.3.1
Ref: https://pypi.org/project/django-graphql-jwt/

Websocket
channels==1.1.8
Ref: https://pypi.org/project/channels/
asgi_redis==1.4.3
Ref: https://pypi.org/project/asgi_redis/
graphql-ws==0.3.1
Ref: https://pypi.org/project/graphql-ws/

kleberbaum committed:
Add bifrost

The bifrost is a Graphql API which creates itself from Django models that are specified
at BIFROST_APPS = {} in the settings.

Currently it works with:

Query:
WagtailPage
AbstractForm
AbstractDocument
AbstractImage
AbstractRendition
BaseSetting
Snippet

Mutation:
AbstractForm

Subscription:
WagtailPage

I am pretty sure this is what comes closest to the holy grail.
No matter what you were looking for, you have found it here.

kleberbaum committed:
Adjust and add images to bifrost

The images model has been adjusted and added
to the bifrost.

kleberbaum committed:
Adjust and add documents to bifrost

The documents model has been adjusted and added
to the bifrost.

kleberbaum committed:
Adjust and add utils to bifrost

The utils model has been adjusted and added
to the bifrost.

kleberbaum committed:
Adjust and add home to bifrost

The home model has been adjusted and added
to the bifrost.

kleberbaum committed:
Format Code

The whole project has been reformatted
using the black formatter.
Ref: https://github.com/psf/black

kleberbaum committed:
Merge branch 'master' into add-bifrost-aa1

# Conflicts:
#	esite/colorfield/forms.py
#	esite/documents/models.py
#	esite/home/models.py
#	esite/images/models.py
#	esite/navigation/templatetags/navigation_tags.py
#	esite/settings/base.py
#	esite/urls.py
#	esite/user/admin.py
#	esite/utils/models.py
#	requirements/base.txt

pinterid commented on esite/urls.py:
```diff
@@ -27,12 +27,15 @@
 
 # Private URLs are not meant to be cached.
 private_urlpatterns = [
+    url(r"", include(api_urls)),
```
Is "r" required?
```suggestion
    url("", include(api_urls)),
```

pinterid reviewed (changes requested):
Just found a single thing

Aichnerc reviewed (approved):
Only looked at comments and general look and feel. I'm no python dev to judge any functionality or coding style. LGTM.

schettn committed:
Update Comments

All commented out source code
is removed now.
Also some comments have been
adjusted.

kleberbaum committed:
Improve code quality

The code quality has been improved due to the codacy check.

schettn reviewed (approved):
LGTM

pinterid reviewed (approved):
LGTM

kleberbaum committed:
Fix bifrost FormPage error

'Page' object has no attribute 'get_form'.

pinterid reviewed (approved):
LGTM

simonprast reviewed (approved):
looks very good to me

schettn reviewed (approved):
lgtm

kleberbaum reviewed (commented):
Was totally ok to merge.
